[2025-10-09T12:08:38.787+0000] {taskinstance.py:1165} INFO - Dependencies all met for <TaskInstance: data_preparation_pipeline_v2.setup_directories manual__2025-10-09T12:08:28.020101+00:00 [queued]>
[2025-10-09T12:08:38.825+0000] {taskinstance.py:1165} INFO - Dependencies all met for <TaskInstance: data_preparation_pipeline_v2.setup_directories manual__2025-10-09T12:08:28.020101+00:00 [queued]>
[2025-10-09T12:08:38.826+0000] {taskinstance.py:1362} INFO - 
--------------------------------------------------------------------------------
[2025-10-09T12:08:38.826+0000] {taskinstance.py:1363} INFO - Starting attempt 1 of 2
[2025-10-09T12:08:38.826+0000] {taskinstance.py:1364} INFO - 
--------------------------------------------------------------------------------
[2025-10-09T12:08:38.873+0000] {taskinstance.py:1383} INFO - Executing <Task(PythonOperator): setup_directories> on 2025-10-09 12:08:28.020101+00:00
[2025-10-09T12:08:38.893+0000] {standard_task_runner.py:55} INFO - Started process 682 to run task
[2025-10-09T12:08:38.905+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'data_preparation_pipeline_v2', 'setup_directories', 'manual__2025-10-09T12:08:28.020101+00:00', '--job-id', '21', '--raw', '--subdir', 'DAGS_FOLDER/data_prep_dag.py', '--cfg-path', '/tmp/tmp5moosqp2']
[2025-10-09T12:08:38.914+0000] {standard_task_runner.py:83} INFO - Job 21: Subtask setup_directories
[2025-10-09T12:08:39.093+0000] {task_command.py:376} INFO - Running <TaskInstance: data_preparation_pipeline_v2.setup_directories manual__2025-10-09T12:08:28.020101+00:00 [running]> on host 0e424a3fb3c0
[2025-10-09T12:08:39.330+0000] {taskinstance.py:1592} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=data_preparation_pipeline_v2
AIRFLOW_CTX_TASK_ID=setup_directories
AIRFLOW_CTX_EXECUTION_DATE=2025-10-09T12:08:28.020101+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2025-10-09T12:08:28.020101+00:00
[2025-10-09T12:08:39.336+0000] {python.py:177} INFO - Done. Returned value was: {'original': PosixPath('/opt/airflow/data/original'), 'clean': PosixPath('/opt/airflow/data/clean'), 'final': PosixPath('/opt/airflow/data/final')}
[2025-10-09T12:08:39.355+0000] {xcom.py:600} ERROR - Could not serialize the XCom value into JSON. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your *** config.
[2025-10-09T12:08:39.357+0000] {taskinstance.py:1851} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/session.py", line 72, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 2385, in xcom_push
    session=session,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/session.py", line 72, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/xcom.py", line 212, in set
    map_index=map_index,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/xcom.py", line 597, in serialize_value
    return json.dumps(value).encode('UTF-8')
  File "/usr/local/lib/python3.7/json/__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)
  File "/usr/local/lib/python3.7/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/local/lib/python3.7/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/usr/local/lib/python3.7/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type PosixPath is not JSON serializable
[2025-10-09T12:08:39.387+0000] {taskinstance.py:1406} INFO - Marking task as UP_FOR_RETRY. dag_id=data_preparation_pipeline_v2, task_id=setup_directories, execution_date=20251009T120828, start_date=20251009T120838, end_date=20251009T120839
[2025-10-09T12:08:39.422+0000] {standard_task_runner.py:105} ERROR - Failed to execute job 21 for task setup_directories (Object of type PosixPath is not JSON serializable; 682)
[2025-10-09T12:08:39.456+0000] {local_task_job.py:164} INFO - Task exited with return code 1
[2025-10-09T12:08:39.532+0000] {local_task_job.py:273} INFO - 0 downstream tasks scheduled from follow-on schedule check
